packages/headless/src/controllers/commerce/search/pagination/headless-search-pagination.ts

```diff
@@ -0,0 +1,31 @@
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {executeSearch} from '../../../../features/commerce/search/search-actions';
+import {commerceSearchReducer as commerceSearch} from '../../../../features/commerce/search/search-slice';
+import {loadReducerError} from '../../../../utils/errors';
+import {
+  buildCorePagination,
+  Pagination,
+} from '../../pagination/core/headless-core-commerce-pagination';
+
+/**
+ * Creates a `Pagination` controller instance.
+ *
+ * @param engine - The headless commerce engine.
+ * @returns A `Pagination` controller instance.
+ * */
+export function buildSearchPagination(engine: CommerceEngine): Pagination {
+  if (!loadSearchPagination(engine)) {
+    throw loadReducerError;
+  }
+
+  return buildCorePagination(engine, {
+    fetchResultsActionCreator: executeSearch,
+  });
+}
+
+function loadSearchPagination(
```

Comment:
If you agree with my idea of using a helper to load the product listing reducer, we could use a similar pattern here.

Reply:
Done in https://github.com/coveo/ui-kit/pull/3475/commits/d1a791410e8384bf85f31c99633036b56aac3f3d